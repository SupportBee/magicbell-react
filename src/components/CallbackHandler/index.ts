import CallbackHandler from './CallbackHandler';
export type { CallbackHandlerProps } from './CallbackHandler';
export default CallbackHandler;
