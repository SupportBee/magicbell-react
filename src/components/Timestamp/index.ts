import Timestamp from './Timestamp';
export type { TimestampProps } from './Timestamp';
export default Timestamp;
