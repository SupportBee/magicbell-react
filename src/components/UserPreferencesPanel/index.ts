import UserPreferencesPanel from './UserPreferencesPanel';
export type { Props as UserPreferencesPanelProps } from './UserPreferencesPanel';
export default UserPreferencesPanel;
