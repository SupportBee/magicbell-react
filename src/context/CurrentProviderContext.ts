import { createContext } from 'react';

export const NO_PROVIDER = 'MAGICBELL_NO_PROVIDER';

const CurrentProviderContext = createContext<String>(NO_PROVIDER);
export default CurrentProviderContext;
