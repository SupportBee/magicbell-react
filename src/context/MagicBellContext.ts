import { createContext, useContext } from 'react';

export interface IMagicBellContext {
  images?: Partial<{
    emptyInboxUrl: string;
  }>;
}

const MagicBellContext = createContext<IMagicBellContext>({} as IMagicBellContext);
export default MagicBellContext;

const useMagicBellContext = () => useContext(MagicBellContext);
export { useMagicBellContext };
