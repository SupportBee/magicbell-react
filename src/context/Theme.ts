export interface IElementTheme {
  backgroundColor: string;
  backgroundOpacity: number;
  borderRadius: string;
  fontFamily: string;
  fontSize: string;
  textAlign: 'left' | 'right' | 'center' | 'justify' | 'initial' | 'inherit';
  textColor: string;
  textTransform: 'none' | 'capitalize' | 'uppercase' | 'lowercase' | 'initial' | 'inherit';
}

export interface IMagicBellTheme {
  icon: {
    borderColor: string;
    width: string;
  };
  unseenBadge: IElementTheme;
  header: IElementTheme;
  container: IElementTheme;
  footer: IElementTheme;
  notification: {
    default: IElementTheme;
    unread: IElementTheme;
    unseen: IElementTheme;
  };
}

const blue = '#3498F4';
const red = '#DF4759';
const black = '#3A424D';

// prettier-ignore
const fontFamily = '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif';
const fontSize = '14px';
const textAlign = 'left';
const textTransform = 'none';

export const defaultTheme = {
  icon: {
    borderColor: blue,
    width: '24px',
  },
  header: {
    backgroundColor: blue,
    backgroundOpacity: 1,
    borderRadius: '8px',
    fontFamily,
    fontSize,
    textAlign,
    textColor: 'white',
    textTransform,
  },
  footer: {
    backgroundColor: blue,
    backgroundOpacity: 1,
    borderRadius: '8px',
    fontFamily,
    fontSize: fontSize,
    textAlign,
    textColor: 'white',
    textTransform,
  },
  unseenBadge: {
    backgroundColor: red,
    backgroundOpacity: 1,
    borderRadius: '2px',
    fontFamily,
    fontSize: '10px',
    textAlign: 'center',
    textColor: 'white',
    textTransform,
  },
  container: {
    backgroundColor: '#FFFFFF',
    backgroundOpacity: 1,
    borderRadius: '8px',
    fontFamily,
    fontSize: '14px',
    textAlign,
    textColor: black,
    textTransform,
  },
  notification: {
    default: {
      backgroundColor: blue,
      backgroundOpacity: 0.1,
      borderRadius: '8px',
      fontFamily,
      fontSize,
      textAlign,
      textColor: black,
      textTransform,
    },
    unread: {
      backgroundColor: blue,
      backgroundOpacity: 0,
      borderRadius: '8px',
      fontFamily,
      fontSize,
      textAlign,
      textColor: black,
      textTransform,
    },
    unseen: {
      backgroundColor: blue,
      backgroundOpacity: 0.05,
      borderRadius: '8px',
      fontFamily,
      fontSize,
      textAlign,
      textColor: black,
      textTransform,
    },
  },
} as IMagicBellTheme;
