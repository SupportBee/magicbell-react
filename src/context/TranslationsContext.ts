import { createContext, useContext } from 'react';

const TranslationsContext = createContext<any>({});
export default TranslationsContext;

const TranslationsProvider = TranslationsContext.Provider;
const useTranslations = () => useContext(TranslationsContext);
export { TranslationsProvider, useTranslations };
