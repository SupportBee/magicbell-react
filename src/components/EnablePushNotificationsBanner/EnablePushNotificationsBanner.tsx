/** @jsx jsx */
import { css, jsx } from '@emotion/react';
import { clientSettings, useConfig } from '@magicbell/react-headless';
import axios from 'axios';
import get from 'lodash/get';
import { useLocalStorage } from 'react-use';
import { useTheme } from '../../context/MagicBellThemeContext';
import { toRGBA } from '../../lib/color';
import { openWindow } from '../../lib/window';
import Text from '../Text';
import CloseBannerButton from './CloseBannerButton';
import EnablePushNotificationsButton from './EnablePushNotificationsButton';

/**
 * Banner to ask the user to enable push notifications.
 *
 * @example
 * <EnablePushNotificationsBanner />
 */
export default function EnablePushNotificationsBanner() {
  const { getState } = clientSettings;
  const { apiKey, userEmail, userExternalId } = getState();
  const config = useConfig();
  const theme = useTheme();
  const { footer: footerTheme } = theme;
  const { notification: notificationTheme } = theme;
  const [wasRequested, setRequestedAt] = useLocalStorage<number | null>(
    `magicbell:${apiKey}:web-push-requested-at`,
    null,
  );
  const isWebPushEnabled = get(config.channels, 'webPush.enabled');

  const enablePushNotifications = () => {
    const baseUrl = get(config.channels, 'webPushNotifications.subscribeUrl');
    const url = axios.getUri({
      url: baseUrl,
      params: {
        user_email: userEmail,
        user_external_id: userExternalId,
        background_color: theme.header.backgroundColor,
        text_color: theme.header.textColor,
      },
    });

    setRequestedAt(Date.now());
    openWindow(url);
  };

  const closeBanner = () => {
    setRequestedAt(Date.now());
  };

  if (wasRequested || !isWebPushEnabled) return null;
  return (
    <div
      css={css`
        padding: 14px 24px !important;
        display: flex;
        text-align: left;
        align-items: center;
        background: ${toRGBA(notificationTheme.unseen.backgroundColor, 0.1)};

        color: ${notificationTheme.unseen.textColor} !important;
        font-family: ${footerTheme.fontFamily} !important;
        text-align: ${footerTheme.textAlign} !important;
        font-size: ${footerTheme.fontSize} !important;

        & > * {
          margin-left: 1rem;
        }
      `}
    >
      <p style={{ flex: 1 }}>
        <Text
          id="web-push-notifications.notice"
          defaultMessage="By enabling browser notifications, youâ€™ll stay up to date even better."
        />
      </p>
      <EnablePushNotificationsButton onClick={enablePushNotifications} />
      <CloseBannerButton onClick={closeBanner} />
    </div>
  );
}
