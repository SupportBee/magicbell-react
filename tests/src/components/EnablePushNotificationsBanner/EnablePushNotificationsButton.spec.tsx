import { fireEvent, render, RenderResult } from '@testing-library/react';
import React from 'react';
import { EnablePushNotificationsButton } from '../../../../src/components/EnablePushNotificationsBanner';
import { MagicBellThemeProvider } from '../../../../src/context/MagicBellThemeContext';
import { defaultTheme } from '../../../../src/context/Theme';

describe('components', () => {
  describe('EnablePushNotificationsButton', () => {
    let onClick: jest.Mock;
    let view: RenderResult;

    beforeEach(() => {
      onClick = jest.fn();

      view = render(
        <MagicBellThemeProvider value={defaultTheme}>
          <EnablePushNotificationsButton onClick={onClick} />
        </MagicBellThemeProvider>,
      );
    });

    afterEach(() => {
      view.unmount();
    });

    describe('render', () => {
      it('renders a styled button', () => {
        expect(view.container).toMatchSnapshot();
      });
    });

    describe('.handleClick', () => {
      it('calls the onClick callback', () => {
        const button = view.getByRole('button');
        fireEvent.click(button);

        expect(onClick).toHaveBeenCalledTimes(1);
      });
    });
  });
});
