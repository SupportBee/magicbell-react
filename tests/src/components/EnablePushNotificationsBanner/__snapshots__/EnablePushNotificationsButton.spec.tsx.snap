// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components EnablePushNotificationsButton render renders a styled button 1`] = `
.emotion-0 {
  padding: 10px!important;
  border-radius: 4px!important;
  background: #3498F4!important;
  color: white!important;
}

<div>
  <button
    class="emotion-0"
  >
    Enable Now
  </button>
</div>
`;
